File: fsharp/difference-of-squares/DifferenceOfSquares.fs
module DifferenceOfSquares

let squareOfSum (number: int): int = pown (List.sum [1..number]) 2

let sumOfSquares (number: int): int = List.sumBy (fun x -> pown x 2) [1..number]

let differenceOfSquares (number: int): int = (squareOfSum number) - (sumOfSquares number) |> abs

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// SquareOfSum returns the square of the sum of the first N natural numbers
func SquareOfSum(n int) int {
	sum := n * (n + 1) / 2
	return sum * sum
}

// SumOfSquares returns the sum of the first N natural numbers squared
func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

// Difference returns the difference between the square of sums and sum of squares
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
